icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java

```diff
@@ -3257,6 +3257,11 @@ public String toLanguageTag() {
         subtags = tag.getExtensions();
         for (String s : subtags) {
             buf.append(LanguageTag.SEP);
+            if (s.endsWith("-true")) {
```

Comment:
Would it not be more logical to do this inside of `LanguageTag.canonicalizeExtension()` instead?